cpp/unreal_plugins/UrdfBot/Source/UrdfParser.h

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include <Eigen/Dense>
+
+#include <CoreMinimal.h>
+
+class FXmlNode;
+
+struct UrdfJointDesc;
+struct UrdfLinkDesc;
+struct UrdfMaterialDesc;
+
+enum class UrdfGeometryType
+{
+    Invalid = 0,
+    Box = 1,
+    Sphere = 2,
+    Cylinder = 3,
+    Mesh = 4,
+};
+
+enum class UrdfJointType
+{
+    Invalid = 0,
+    Revolute = 1,
+    Continuous = 2,
+    Prismatic = 3,
+    Fixed = 4,
+    Floating = 5,
+    Planar = 6
+};
+
+enum class CalibrationType
+{
+    Invalid = 0,
+    Rising = 1,
+    Falling = 2
+};
+
+struct UrdfGeometryDesc
+{
+    UrdfGeometryType type_;
+    FVector size_ = FVector(0.07); // required by Box, in [m]
```

Comment:
- default values shouldn't be specific to a particular robot system
- defaults should be 0.0f and should be defined for all primitive data types

Reply:
fixed